core/src/main/java/com/linecorp/armeria/server/ServerConfig.java

```diff
@@ -169,10 +176,22 @@
                                    gracefulShutdownQuietPeriod, "gracefulShutdownQuietPeriod");
 
         requireNonNull(blockingTaskExecutor, "blockingTaskExecutor");
-        blockingTaskExecutor =
-                ExecutorServiceMetrics.monitor(meterRegistry, blockingTaskExecutor,
-                                               "blockingTaskExecutor", "armeria");
+        final ScheduledExecutorService unwrappedBlockingTaskExecutor;
+        if (blockingTaskExecutor instanceof BlockingTaskExecutor) {
+            unwrappedBlockingTaskExecutor = ((BlockingTaskExecutor) blockingTaskExecutor).unwrap();
+        } else {
+            unwrappedBlockingTaskExecutor = blockingTaskExecutor;
+        }
+
+        new ExecutorServiceMetrics(
+                unwrappedBlockingTaskExecutor,
+                "blockingTaskExecutor", "armeria", ImmutableList.of())
+                .bindTo(meterRegistry);
+        blockingTaskExecutor = new TimedScheduledExecutorService(meterRegistry, blockingTaskExecutor,
+                                                                 "blockingTaskExecutor", "armeria.",
+                                                                 ImmutableList.of());
         this.blockingTaskExecutor = UnstoppableScheduledExecutorService.from(blockingTaskExecutor);
```

Comment:
How about extracting this into a private static method, so that the constructor code is more readable, i.e.:

```java
        ...
        this.blockingTaskExecutor = monitorBlockingTaskExecutor(blockingTaskExecutor);
        ...
    }

    private static ScheduledExecutorService monitorBlockingTaskExecutor(ScheduledExecutorService executor) {
        final ScheduledExecutorService unwrapped;
        if (executor instanceof BlockingTaskExecutor) {
            unwrapped = ((BlockingTaskExecutor) executor).unwrap();
        } else {
            unwrapped = executor;
        }
        ...
        return UnstoppableScheduledExecutorService.from(...);
    }
```

Reply:
I reflected it. Please check the modifications again 😀